src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeProcessTests.cs

```diff
@@ -57,6 +58,70 @@ public void Start_FailsToFindNode_FileNotFoundException()
             act.Should().ThrowExactly<FileNotFoundException>().And.Message.Should().Contain("node.exe");
         }
 
+        [TestMethod]
+        public async Task Start_SecondCall_ProcessFailed_StartsNewProcess()
+        {
+            var fakeNodeExePath = CreateScriptThatPrintsPortNumber(123, isHanging: true);
+            var startupScriptPath = "dummy path";
+
+            var nodeLocator = SetupNodeLocator(fakeNodeExePath);
+            var testSubject = CreateTestSubject(startupScriptPath, nodeLocator: nodeLocator.Object);
+
+            try
+            {
+                await testSubject.Start();
+
+                var oldProcess = testSubject.Process;
+
+                oldProcess.HasExited.Should().BeFalse();
+                oldProcess.Kill();
+                await oldProcess.WaitForExitAsync();
+                oldProcess.HasExited.Should().BeTrue();
+
+                await testSubject.Start();
+
+                var newProcess = testSubject.Process;
+                newProcess.Should().NotBeSameAs(oldProcess);
+                newProcess.HasExited.Should().BeFalse();
+                newProcess.Kill();
+            }
+            finally
+            {
+                // Kill spawned process
+                testSubject.Dispose();
```

Comment:
This is indirect cleanup; it's relying on the testSubject being correctly implemented and doing the cleanup for us. The test doesn't need to do that - it created the test processes, so it can ensure that they are cleaned up by killing/disposing them directly.
Same for the other test.